circuit/types/integers/src/add_checked.rs

```diff
@@ -73,37 +73,38 @@ impl<E: Environment, I: IntegerType> AddChecked<Self> for Integer<E, I> {
                 Some(value) => Integer::constant(console::Integer::new(value)),
                 None => E::halt("Integer overflow on addition of two constants"),
             }
-        } else {
-            // TODO: Optimize
+        } else if I::is_signed() {
             // Instead of adding the bits of `self` and `other` directly, the integers are
             // converted into a field elements, and summed, before converting back to integers.
             // Note: This is safe as the field is larger than the maximum integer type supported.
             let sum = self.to_field() + other.to_field();
 
-            // Extract the integer bits from the field element, with a carry bit.
-            let (sum, carry) = match sum.to_lower_bits_le(I::BITS as usize + 1).split_last() {
-                Some((carry, bits_le)) => (Integer::from_bits_le(bits_le), carry.clone()),
+            // Extract the integer bits from the field element, ignoring the carry bit as it is not relevant for signed addition.
+            let sum = match sum.to_lower_bits_le(I::BITS as usize + 1).split_last() {
+                Some((_, bits_le)) => Integer::from_bits_le(bits_le),
```

Comment:
```suggestion
                Some(bits_le) => Integer::from_bits_le(bits_le),
```
To match suggested change above.

Reply:
See comment above.